src/components/app/App.js

```diff
@@ -26,11 +50,11 @@ class App extends Component {
   render() {
     return (
       <main>
-        {
-          this.state.selectedMovie ? 
-          <MoviePage details={this.state.selectedMovie} viewHome={this.viewHome}/> : 
-          <Movies allMovies={this.state.allMovies} viewMovie={this.viewSelectedMovie} />
-        }
+        {this.state.error && <h3>Oops, something went wrong!</h3>}
+        {this.state.allMoviesLoading && <img src="https://i.gifer.com/ZKZg.gif" />}
+        {this.state.selectedMovie 
+        ? (!this.state.singleMovieLoading && <MoviePage details={this.state.selectedMovie} viewHome={this.viewHome}/>)
```

Comment:
Lots of `this.state.___` in this render. Maybe it could be cleaner if we used destructuring?